server/lib/wsapi/chat.ts

```diff
@@ -168,29 +143,17 @@ export class ChatApi {
     }),
     'getUser',
     throttleMiddleware(retrievalThrottle, (data: Map<string, any>) => data.get('user')),
-    'getChannel',
   )
   async getHistory(data: Map<string, any>) {
-    const { limit, beforeTime } = data.get('body')
     const user = data.get('user')
-    const channel = data.get('channel')
-    // TODO(tec27): lookup channel keys case insensitively?
-    if (!this.state.users.has(user.name) || !this.state.users.get(user.name)!.has(channel)) {
-      throw new errors.Forbidden('must be in a channel to retrieve message history')
-    }
+    const { channel, limit, beforeTime } = data.get('body')
 
-    const messages = await getMessagesForChannel(
-      channel,
-      user.session.userId,
-      limit,
-      beforeTime && beforeTime > -1 ? new Date(beforeTime) : undefined,
-    )
-    return messages.map(m => ({
-      id: m.msgId,
-      user: m.userName,
-      sent: +m.sent,
-      data: m.data,
-    }))
+    try {
+      const chatService = container.resolve(ChatService)
+      return chatService.getChannelHistory(channel, user.name, limit, beforeTime)
+    } catch (err) {
+      return convertChatServiceError(err)
```

Comment:
not gonna ask you to fix this either, because this is moving to HTTP, but in the future I think it's preferable to make this a `throw` to make the semantics clear. This works now because `convertChatServiceError` always throws, but looking at this code in isolation, this sure looks like it's returning a success response